### DISCUSSION TITLE
Need help getting set up with package

### DISCUSSION THREAD

**Discussion by Supernova3339** [cjmellor/level-up#69] [Q&A] [answered]
Whenever  I try `$user->addPoints(10);`, it throws an error.

```
SQLSTATE[23000]: Integrity constraint violation: 1048 Column 'user_id' cannot be null
INSERT INTO `experiences` (`level_id`, `experience_points`, `user_id`, `updated_at`, `created_at`) VALUES (1, 10, ?, 2024-03-31 15:02:23, 2024-03-31 15:02:23)
```

How can I fix this? I have all of the imports in the User model ( GiveExperience, GiveAchievement )
My configuration is unchanged as well and migrations went fine. I have seeded 100 levels and created 1 achievement.

**Comment by cjmellor** [comment]
Hi,

I'm afraid I can't replicate this. I just did a fresh install and it is working as expected.

`GiveAchievement` isn't a trait, it is `HasAchievements` -- mistype?

The error implies the User you're trying to assign the points to doesn't exist. Make sure `$user` returns a User model.

**Reply by Supernova3339** [reply, marked as answer]
just checked.

I have "GiveExperience" and "HasAchievements".

I have solved this issue on my own, it was me being dumb with how I was fetching my user.

I was doing (User $user)

when I should have been doing...
```php
$user = User::find(Auth::id())
```

Super appreciate your response @cjmellor ❤️
